src/auth/credential.ts

```diff
@@ -186,75 +185,59 @@ export interface GoogleOAuthAccessToken {
 }
 
 /**
- * A wrapper around the http and https request libraries to simplify & promisify JSON requests.
- * TODO(inlined): Create a type for "transit".
+ * Obtain a new OAuth2 token by making a remote service call.
  */
-function requestAccessToken(transit, options: object, data?: any): Promise<GoogleOAuthAccessToken> {
-  return new Promise((resolve, reject) => {
-    const req = transit.request(options, (res) => {
-      const buffers: Buffer[] = [];
-      res.on('data', (buffer) => buffers.push(buffer));
-      res.on('end', () => {
-        try {
-          const json = JSON.parse(Buffer.concat(buffers).toString());
-          if (json.error) {
-            let errorMessage = 'Error fetching access token: ' + json.error;
-            if (json.error_description) {
-              errorMessage += ' (' + json.error_description + ')';
-            }
-            reject(new FirebaseAppError(AppErrorCodes.INVALID_CREDENTIAL, errorMessage));
-          } else if (!json.access_token || !json.expires_in) {
-            reject(new FirebaseAppError(
-              AppErrorCodes.INVALID_CREDENTIAL,
-              `Unexpected response while fetching access token: ${ JSON.stringify(json) }`,
-            ));
-          } else {
-            resolve(json);
-          }
-        } catch (err) {
-          reject(new FirebaseAppError(
-            AppErrorCodes.INVALID_CREDENTIAL,
-            `Failed to parse access token response: ${err.toString()}`,
-          ));
-        }
-      });
-    });
-    req.on('error', reject);
-    if (data) {
-      req.write(data);
+function requestAccessToken(client: HttpClient, request: HttpRequestConfig): Promise<GoogleOAuthAccessToken> {
+  return client.send(request).then((resp) => {
+    const json = resp.data;
+    if (json.error) {
+      let errorMessage = 'Error fetching access token: ' + json.error;
+      if (json.error_description) {
+        errorMessage += ' (' + json.error_description + ')';
+      }
+      throw new FirebaseAppError(AppErrorCodes.INVALID_CREDENTIAL, errorMessage);
+    } else if (!json.access_token || !json.expires_in) {
+      throw new FirebaseAppError(
+        AppErrorCodes.INVALID_CREDENTIAL,
+        `Unexpected response while fetching access token: ${ JSON.stringify(json) }`,
+      );
+    } else {
+      return json;
     }
-    req.end();
+  }).catch((err) => {
+    throw new FirebaseAppError(
+      AppErrorCodes.INVALID_CREDENTIAL,
+      `Failed to parse access token response: ${err.toString()}`,
+    );
   });
 }
 
 /**
  * Implementation of Credential that uses a service account certificate.
  */
 export class CertCredential implements Credential {
-  private certificate_: Certificate;
+  private readonly certificate_: Certificate;
```

Comment:
since you're already modifying this, let's remove the trailing underscore.

Reply:
Done